pallets/elections/src/lib.rs

```diff
@@ -392,24 +418,14 @@ pub mod pallet {
             Ok(())
         }
 
-        fn kick_out_underperformed_non_reserved_validators() {
-            let mut to_be_kicked = BTreeMap::from_iter(ToBeKickedOutFromCommittee::<T>::iter());
-            if !to_be_kicked.is_empty() {
-                let mut to_be_removed = Vec::new();
-                let mut filtered_non_reserved_in_order = Vec::new();
-                let non_reserved_validators = NextEraNonReservedValidators::<T>::get();
-                for non_reserved_validator in non_reserved_validators {
-                    match to_be_kicked.remove_entry(&non_reserved_validator) {
-                        Some(candidate_and_reason) => to_be_removed.push(candidate_and_reason),
-                        None => filtered_non_reserved_in_order.push(non_reserved_validator),
-                    }
-                }
-                if !to_be_removed.is_empty() {
-                    info!(target: "pallet_elections", "Removing following validators from non reserved, {:?}", to_be_removed);
-                    NextEraNonReservedValidators::<T>::put(filtered_non_reserved_in_order);
-                    Self::deposit_event(Event::KickOutValidators(to_be_removed));
-                }
-                let _result = ToBeKickedOutFromCommittee::<T>::clear(u32::MAX, None);
+        fn emit_fresh_bans_event() {
+            let active_era = <T as Config>::EraInfoProvider::active_era().unwrap_or(1);
+            let fresh_bans = Banned::<T>::iter()
+                .filter(|(_acc, info)| info.start == active_era)
```

Comment:
Actually, when the elections happen for era `N`, the `active_era` is still `N-1` (this is because of how pallet staking works), so we need to compare with `active_era + 1` here, as now even is emitted, but one era later than it should be

Reply:
Yes, my PR introduced this error. Was not sure about whether it is already `N` or still `N-1`

Reply:
Huh, I was pretty sure I fixed that :thinking: